### ISSUE TITLE
Audio and Video go out of sync when using video-sync=display-resample

### ISSUE THREAD

**Issue by meta3ark** [mpv-player/mpv#14141]
### Important Information

- mpv v0.37.0-787-gab504514
- Windows 10 Pro
- [Shinchiro](https://github.com/shinchiro/mpv-winbuild-cmake/releases/download/20240416/mpv-x86_64-20240416-git-ab50451.7z)
- Change of behaviour between 22/10/2023 and 29/10/2023 shinchiro sourceforge builds
- Nvidia GTX 1060 6GB, 551.86

### Reproduction steps

Play a local 23.976Hz mkv file on a frame rate matched TV for about 10 minutes.

mpv --no-config --vo=gpu-next --video-sync=display-resample

### Expected behavior
Audio/Video does not go out of sync after a period of time

### Actual behavior
After approximately 7 minutes I observed mis-timed frames in stats and the audio (lip-sync) goes noticably out of sync with the video.

### Log file
[video-sync-display-resample-log.txt](https://github.com/mpv-player/mpv/files/15306955/video-sync-display-resample-log.txt)
[video-sync-audio-log.txt](https://github.com/mpv-player/mpv/files/15306956/video-sync-audio-log.txt)

### Sample files

Any 23.976hz mkv file

### Additional Info

using [Revert "d3d11: calc vsync interval on real stats...](https://github.com/kasper93/mpv/actions/runs/8402221581) fixes the problem
related issue #13952 and possibly #13439

**Comment by kasper93** [comment]
> using [Revert "d3d11: calc vsync interval on real stats...](https://github.com/kasper93/mpv/actions/runs/8402221581) fixes the problem

Ok the master now. Reopen if you see other issues.

**Comment by Jules-A** [comment]
I wasn't getting this before but it looks like I've been seeing this (very slightly) in the past few weeks though it could be something else at fault.

**Comment by kasper93** [comment]
> I thought it was fixed but it is still happening but now every approx 20 minutes. A very noticable video stutter/jerk. Stats -> 1 delayed frame. It resumes and video/audio do appear to be in sync, so better than originally.
> 
> Issue does not occur after reverting to October 2023 build and changing nothing else.

Find the regression range, we will not search for bugs that only you can reproduce and not reliably at best.

**Comment by kasper93** [comment]
So you confirm that this build is
good: https://nightly.link/mpv-player/mpv/actions/artifacts/1603867417.zip
bad: https://nightly.link/mpv-player/mpv/actions/artifacts/1603005534.zip

**Comment by Jules-A** [comment]
> So you confirm that this build is [good](https://nightly.link/mpv-player/mpv/actions/artifacts/1603867417.zip) and this is [bad](https://nightly.link/mpv-player/mpv/actions/artifacts/1603005534.zip)

For me they both had issues still, however the one you marked as good is better when seeking.

**Comment by kasper93** [comment]
@meta3ark: Can you provide regression range for the issue you are seeing?

@Jules-A: Could you report an issue, with all required info, as your problem is not related to the changes discussed here.

**Comment by Jules-A** [comment]
> I tried good/bad and latest github action build. All working without delayed frames/stutters. Nothing obvious I can see and the 20 min plus testing periods make finding a reproducible scenario difficult. Closing this as it's a different issue i think.

Hmm... That is odd, both those afaik the "good" build was between those 4 commits you listed and the "bad" build was before them? (I think labeling was wrong way around).
Also yeah, it's not easy to test and possibly is another issue but I don't really have time to investigate it more and report myself right now. (I suspected it wasn't actually the MPV commits at fault and if that's the case is it really an MPV issue in the first place?).
Even without the dropped/delayed frames randomly occurring, the normal Vsync jitter is doubled for me in the newer builds though it's still so low that alone shouldn't be causing any issues.

**Comment by kasper93** [comment]
> I tried good/bad and latest github action build. All working without delayed frames/stutters.

Latest, you mean for current master branch?

> Libmpv builds from 15/06 onwards have stutters/delays.

By libmpv build you mean the build from shinchiro?

If our CI build works fine, it is likely not a mpv issue. Nor it is ffmpeg or libplacebo one, because we build them too from master branches. So the issue if I understand correctly is likely somewhere else. Let us know if you find something.

Since the regression is pretty new, you have build for each commit on our CI, if you click on green tick mark. Also shinchiro and zhongfly build are there, so it should be possible to find the regression range.

**Comment by Jules-A** [comment]
> Since the regression is pretty new, you have build for each commit on our CI, if you click on green tick mark. Also shinchiro and zhongfly build are there, so it should be possible to find the regression range.

According to their replies they had the same regression range as me which I posted 3 days ago
>Good: https://github.com/zhongfly/mpv-winbuild/releases/tag/2024-06-14-a41cdf0
Bad: https://github.com/zhongfly/mpv-winbuild/releases/tag/2024-06-15-6c90836

> I then tested versions suggested by Jules and can confirm one of the 4 commits listed is the cause.

**Comment by kasper93** [comment]
> It would be nice if it was part of the CI builds by default.

It is quite annoying to configure shared library build, with static libraries dependencies in Meson, that's why it is not available. And likely this will not change until this is implemented on Meson side https://github.com/mesonbuild/meson/pull/13086#discussion_r1576818094

**Comment by kasper93** [comment]
> but had some with the shinchiro build

Would be really nice if you could track down the issue, with shinchiro build. I think a lot of people are affected/using those ones.

**Comment by kasper93** [comment]
Sure, no worries. If it works for you it is fine. Thanks for trying to figure it out in the first place.

**Comment by Jules-A** [comment]
> including 3rdpary libs versions
Is there an easy way to see all the commits from all 3rd party libs?

**Comment by kasper93** [comment]
> Is there an easy way to see all the commits from all 3rd party libs between a specific time period?

Go to build details, and click on `Packages Version`. I don't see anything of importance and if those changes really regress the stability of the playback, we need to diagnose what is really going on. I don't know if you @Jules-A have easier way to reproduce the issue, but you could try doing custom builds with reverted 3rdparty libs. We need to narrow down it step by step, no other way, I think.

**Comment by Jules-A** [comment]
> Go to build details, and click on `Packages Version`. 

Ah I was hoping for something easier, it's still hard to check that way and unless I'm mistaken, the only thing I can see that might affect things would be https://github.com/libass/libass/pull/780 ?
I don't really have time to set up building / reverting stuff manually sorry. 
I may be mistaken but I think the drops when running on my usual config were when it was caching (CPU use is increased quite a bit when it occurs). Maybe increasing cache sizes can help reproduce, otherwise maybe using a computer with a slower CPU?

**Comment by kasper93** [comment]
> Ah I was hoping for something easier, it's still hard to check that way and unless I'm blind or not understanding the changes, the only thing I can see that might affect things would be https://github.com/libass/libass/pull/780 ?

It is quite literally impossible for this change to regress performance or make any significant change in normal use of libass. I agree it looks suspicious in this regression range, but I cannot see how it can create a problem. Also @meta3ark confirmed that his local build is not affected, while it also includes this change you are asking about. Also CI builds are not affected.

Do you see a difference between v3 and "normal" build?

**Comment by kasper93** [comment]
@zhongfly: would you be possible for you to rerun this build without libass version change? Just for the sake of eliminating this factor. I don't follow how to build those builds with GH actions.

**Comment by zhongfly** [comment]
> @zhongfly: would you be possible for you to rerun this build without libass version change?

Unfortunately, it can't. Even building the exact same version again is difficult. This is because each run is based on the latest cache and updates all git repositories before starting building.

If necessary, I can build one where all dependencies are up to date, but revert libass to [libass@306bf76](https://github.com/libass/libass/commit/306bf76ca5ac05d936614e5c2ffc23aeac6dfb63).

**Comment by kasper93** [comment]
> If necessary, I can build one where all dependencies are up to date, but reset libass to [libass@306bf76](https://github.com/libass/libass/commit/306bf76ca5ac05d936614e5c2ffc23aeac6dfb63). https://github.com/zhongfly/mpv-winbuild/actions/runs/9859120502

I guess that works, thanks. @Jules-A could you try this build?

**Comment by Jules-A** [comment]
> I guess that works, thanks. @Jules-A could you try this build?

Thanks zhongfly, I will test this tonight.

>Do you see a difference between v3 and "normal" build?

I didn't end up testing the non-v3 builds again enough to tell (just ~15mins but I didn't see any issues in that time). Originally I did test it when it first occurred and it didn't make a diff then though.

**Comment by Jules-A** [comment]
Okay, I also tested master for another 2hrs30mins and wasn't able to reproduce... Now I'm questioning if it was even the program itself rather than some background process I was unaware of (I also noticed this time I didn't have the mouse mouse randomly reappearing for the whole time I was testing). Only thing I did differently this time was update Windows 10 to latest monthly update and delete both AMD's and MPV's cache.
There's still a lot of dropped/mistimed frames when MPV isn't in foreground but pretty sure that's a separate issue so I think I might call it quit testing and wait on some other synchronization fixes to land... Sorry guys that I wasn't able to be of more help.
